src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -245,6 +259,33 @@ private function canUserHideContent(Location $location): bool
             [$target]
         );
     }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     */
+    private function isPreviewable(
+        Location $location,
+        Content $content,
+        string $languageCode
+    ): bool {
+        $versionNo = $content->getVersionInfo()->versionNo;
+
+        $siteAccesses = $this->siteaccessResolver->getSiteAccessesListForLocation(
+            $location,
+            $versionNo,
+            $languageCode
+        );
```

Comment:
To me, it's a bit ineffective to fetch the full list only to check if it's empty (even if you only moved the code). If there is no dedicated API method I would first check if `$canPreview` is `true` before fetching SiteAccesses list.